File: generated-code/codecraft/go/model/vec2_int32.go
package model

import "io"
import . "trans_gen_test/stream"

type Vec2Int32 struct {
    X int32
    Y int32
}

func NewVec2Int32(x int32, y int32) Vec2Int32 {
    return Vec2Int32 {
        X: x,
        Y: y,
    }
}

func ReadVec2Int32(reader io.Reader) Vec2Int32 {
    var x int32
    x = ReadInt32(reader)
    var y int32
    y = ReadInt32(reader)
    return Vec2Int32 {
        X: x,
        Y: y,
    }
}

func (vec2Int32 Vec2Int32) Write(writer io.Writer) {
    x := vec2Int32.X
    WriteInt32(writer, x)
    y := vec2Int32.Y
    WriteInt32(writer, y)
}

File: generated-code/pascal/codecraft/FileReadWrite/Vec2Int32.pas
unit Vec2Int32;

interface

uses
    Stream,
    SysUtils;

type
    // 2 dimensional vector.
    TVec2Int32 = class
        // `x` coordinate of the vector
        x: Int32;
        // `y` coordinate of the vector
        y: Int32;
        constructor Create(x: Int32; y: Int32);
        // Read Vec2Int32 from input stream
        class function ReadFrom(stream: TStream): TVec2Int32; static;
        // Write Vec2Int32 to output stream
        procedure WriteTo(stream: TStream);
        function ToString: ansistring; override;
    end;

implementation

constructor TVec2Int32.Create(x: Int32; y: Int32);
begin
    self.x := x;
    self.y := y;
end;

class function TVec2Int32.ReadFrom(stream: TStream): TVec2Int32;
var x: Int32;
var y: Int32;
begin
    x := stream.ReadInt32;
    y := stream.ReadInt32;
    result := TVec2Int32.Create(x, y);
end;

procedure TVec2Int32.WriteTo(stream: TStream);
begin
    stream.WriteInt32(x);
    stream.WriteInt32(y);
end;

function TVec2Int32.ToString: ansistring;
begin
    result := 'Vec2Int32 {';
    result += 'x=';
    result += IntToStr(x);
    result += ', ';  
    result += 'y=';
    result += IntToStr(y);
    result += '}';
end;

end.